test/infoSpec.js

```diff
@@ -129,6 +130,133 @@ describe('Info', () => {
     })
   })
 
+  describe('authentication', () => {
+    beforeEach(async function () {
+      this.newSocket = new MockSocket()
+      await this.plugin.addSocket(this.newSocket)
+    })
+
+    it('should deny an authentication request with wrong method', async function () {
+      this.newSocket.emit('message', btpPacket.serializeFulfill({
+        transferId: 'b38a5203-bdb8-f11f-db01-5a32cf1a4e43',
+        fulfillment: 'Ndr_HMuLPPl0idUlvAXFXBVQTFOizq-nXozej0KIA7k'
+      }, 100, []))
+
+      await new Promise((resolve) => {
```

Comment:
You don't need to wrap this in a promise. `this.mockSocket.isDone()` makes sure all handlers registered via `.reply()` have been called.